django/forms/widgets.py

```diff
@@ -893,3 +897,134 @@ def __init__(self, attrs=None, date_format=None, time_format=None):
         super(SplitHiddenDateTimeWidget, self).__init__(attrs, date_format, time_format)
         for widget in self.widgets:
             widget.input_type = 'hidden'
+
+
+class SelectDateWidget(Widget):
+    """
+    A Widget that splits date input into three <select> boxes.
+
+    This also serves as an example of a Widget that has more than one HTML
+    element and hence implements value_from_datadict.
+    """
+    none_value = (0, '---')
+    month_field = '%s_month'
+    day_field = '%s_day'
+    year_field = '%s_year'
+
+    date_re = re.compile(r'(\d{4})-(\d\d?)-(\d\d?)$')
+
+    def __init__(self, attrs=None, years=None, months=None, empty_label=None):
+        self.attrs = attrs or {}
+
+        # Optional list or tuple of years to use in the "year" select box.
+        if years:
+            self.years = years
+        else:
+            this_year = datetime.date.today().year
+            self.years = range(this_year, this_year + 10)
```

Comment:
To be consistent with the rest of the codebase, I'd import `from django.utils.six.moves import range` first.